doc/poe/poe_hld.md

```diff
@@ -0,0 +1,370 @@
+  - PoE power, current and voltage;
+  - PSE information;
+- apply PoE controller configuration;
+- track changes in config, appl DB;
+
+##### Configuration
+
+Below is a example of PoE configuration used by PoE manager on start up. The mapping between PoE port and ASIC port is done via front panel id.
+```
+{
+  "hw_info": "integrated_mcu",
+  "pse_list": [
+    {
+      "pse_index": 0
+    },
+    {
+      "pse_index": 1
+    },
+    ...
+  ],
+  "port_mapping_list": [
+    {
+      "Ethernet0": 
+      {
+          "front_panel_index": 1
+      },
+      ...
+      "Ethernet47": 
+      {
+          "front_panel_index": 48
+      }
+    }
+  ]
+}
+```
+
+#### LLDP manager (modified)
+
+To exchange the PoE information with peer about power supply capability, the LLDP protocol is used. In case to support
+that, LLDP manager is required to be modified as following:
+
+- LLDP manager gets PoE information (from application PoE database).
+- LLDP manager adds LLDP TLV with PoE information into LLDP packet.
+- LLDP negotiation (future versions).
+
+### High-Level Design 
+
+This section covers the high level design of the built-in SONiC PoE feature.
+
+This section covers the following points in detail.
+
+	- Existing moduels are not modified by this design.
+	- Repositories that will be changed: sonic-buildimage, ...
+	- Module/sub-module interfaces and dependencies. 
+	- SWSS and Syncd are not changed.
+	- This design doesn't change existing DB Schema, only new are added.
+	- Linux dependencies and interface
+	- Fastboot requirements & dependencies
+	- Scalability and performance requirements/impact
+	- Memory requirements
+	- Docker dependency
+	- Build dependency if any
+	- Management interfaces - SNMP, CLI, RestAPI, etc.,
+	- Serviceability and Debug (logging, counters, trace etc) related design
+	- Is this change specific to any platform? Are there dependencies for platforms to implement anything to make this feature work? If yes, explain in detail and inform community in advance.
+	- SAI API requirements, CLI requirements, ConfigDB requirements. Design is covered in following sections.
+
+##### PoE flows
+
+###### PoE data collection
+
+"Show" flow:
+  1. Get PoE controller information;
+  2. Update application DB with the PoE status;
+  3. Repeat step 1-2 each seconds (optional);
+  4. SONiC CLI read and display PoE infomation from application DB;
+
+![sequence_show_command](images/poe_show_flow.png)
+
+###### PoE configuration
+
+Configuration flow:
+  1. Configure PoE using SONiC CLI;
+  2. Update Config DB accordingly;
+  3. Poemgrd subscribe to related tables in Config DB;
+  4. Call PoE AI for related configuration;
+
+![sequence_cfg_command](images/poe_config_flow.png)
+
+
+### SAI API 
+
+- SAI object types extended to support PoE
+- SAI API list is extended to supprot PoE
+- New PoE switch type added
+
+### PoE Abstraction Interface
+
+The PoE Manager uses a new SAI PoE library that implements the PoE Abstraction Interface.
```

Comment:
can you share the poe sai library pr? is it already merged in sai?

Reply:
SAI PR -https://github.com/opencomputeproject/SAI/pull/1977 (merged)
Implementation will be (PoE HW) vendor specific, we will be providing an example, working implementation for MicroChip PSE (in development)